### PR TITLE
set request _locale attribute breaks the switcher.

### PR DESCRIPTION
**Pull request by nnmer**
to reproduce:
Let assume we didn't comment proposed PR
Add to LocaleController->switchAction after setting $redirectToRoute next lines :

```
dump($request);
dump($_locale);
die();
```

config settings next:

```
lunetics_locale:
    strict_mode: true
    allowed_locales:
        - zh
        - en
    guessing_order:
        - session
        - cookie
        - browser
        - query
        - router
    switcher:
        use_controller: true
    cookie:
        set_on_change: true
```

and try to change the locale with switcher. Assume we changing the locale by next url: ..../changeLocale/cc
the result will be:

```
LocaleController.php on line 61:
Request {#7 ▼
  +attributes: ParameterBag {#10 ▶}
  +request: ParameterBag {#8 ▶}
  +query: ParameterBag {#9 ▶}
  +server: ServerBag {#13 ▶}
  +files: FileBag {#12 ▶}
  +cookies: ParameterBag {#11 ▶}
  +headers: HeaderBag {#14 ▶}
  #content: null
  #languages: array:2 [▶]
  #charsets: null
  #encodings: null
  #acceptableContentTypes: null
  #pathInfo: "/changeLocale/cc"
  #requestUri: "/changeLocale/cc"
  #baseUrl: ""
  #basePath: null
  #method: "GET"
  #format: null
  #session: Session {#504 ▶}
  #locale: "en"
  #defaultLocale: "zh_CN"
}
LocaleController.php on line 62:
"en"
```

SO, locale change was not picked,
but if we do comment proposed PR, the result for the same query is:

```
LocaleController.php on line 61:
Request {#7 ▼
  +attributes: ParameterBag {#10 ▶}
  +request: ParameterBag {#8 ▶}
  +query: ParameterBag {#9 ▶}
  +server: ServerBag {#13 ▶}
  +files: FileBag {#12 ▶}
  +cookies: ParameterBag {#11 ▶}
  +headers: HeaderBag {#14 ▶}
  #content: null
  #languages: array:2 [▶]
  #charsets: null
  #encodings: null
  #acceptableContentTypes: null
  #pathInfo: "/changeLocale/cc"
  #requestUri: "/changeLocale/cc"
  #baseUrl: ""
  #basePath: null
  #method: "GET"
  #format: null
  #session: Session {#504 ▶}
  #locale: "cc"
  #defaultLocale: "zh_CN"
}
LocaleController.php on line 62:
"cc"
```

### CODE REVIEW

**Comment by nnmer** [comment]
The second commit - with the same configuration:
```
lunetics_locale:
    strict_mode: true
    allowed_locales:
        - zh
        - en
    guessing_order:
        - session
        - cookie
        - browser
        - query
        - router
    switcher:
        use_controller: true
    cookie:
        set_on_change: true
```

File: EventListener/LocaleListener.php [hunk 1]
```diff
@@ -110,7 +110,7 @@ public function onKernelRequest(GetResponseEvent $event)
         if ($locale) {
             $this->logEvent('Setting [ %s ] as locale for the (Sub-)Request', $locale);
             $request->setLocale($locale);
-            $request->attributes->set('_locale', $locale);
+//            $request->attributes->set('_locale', $locale);
```
Comment (ikwattro) [inline EventListener/LocaleListener.php]: remove commented code

**Comment by ikwattro** [comment]
Please rebase on master and check failing tests. Thanks!

**Comment by goetas** [comment]
i'm closing this, 7 years passed :(

**Pull request closed without merging**